pkgs/by-name/py/pysolfc/package.nix

```
@@ -0,0 +1,108 @@
+{
+  lib,
+  callPackage,
+  desktop-file-utils,
+  fetchzip,
+  freecell-solver,
+  python3Packages,
+  testers,
+  # Configurable options
+  withFreecellSolver ? true,
+  withPillow ? true,
+  withPygame ? true,
+}:
+
+let
+  _sources = callPackage ./sources.nix { };
+  self = python3Packages.buildPythonApplication {
+    inherit (_sources.pysolfc) pname version src;
+    pyproject = true;
+
+    patches = [
+      ./0000-pysolfc-datadir.patch
+    ];
+
+    dependencies = with python3Packages; ([
+      attrs
+      configobj
+      pycotap
+      pysol-cards
+      random2
+      six
+      tkinter
+    ] ++ lib.optionals withPillow [
+      pillow
+    ] ++ lib.optionals withPygame [
+      pygame
+    ]);
+
+    build-system = [ python3Packages.setuptools ];
+
+    nativeBuildInputs = [ desktop-file-utils ];
+
+    propagatedBuildInputs = lib.optionals withFreecellSolver [
+      freecell-solver
+    ];
+
+    # No tests in archive
+    doCheck = false;
+
+    postPatch = ''
+      desktop-file-edit \
+        --set-key Icon \
+        --set-value ${placeholder "out"}/share/icons/pysol01.png data/pysol.desktop
+    '';
+
+    preBuild = ''
+      export OLD_PYTHONPATH=''${PYTHONPATH}
```

```suggestion
      export OLD_PYTHONPATH=$PYTHONPATH
```
we can avoid quoting hell by not using ${}

---

`${}` is a bit safer for reading vars in Bash.

---

Well, you need to use it when you do advanced things with variables and if they contain certain characters. Quoting is usually more important but since the content is expected and trust worthy we don't need to worry as much.

---

OK. Done.